### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by elishkaflint**
Still have 2 controller tests which I need to fix but would be good to get feedback on the rest in the meantime

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -37,6 +37,7 @@ group :development, :test do
   gem 'rails-controller-testing'
   gem 'rspec-rails', '~> 3.5'
   gem 'selenium-webdriver'
+  gem 'timecop'
```
Comment (dct-lau17) [inline Gemfile]: remove this?

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Post < ApplicationRecord
+
+  belongs_to :users, class_name: 'User', :foreign_key => "user_id"
```
Comment (dct-lau17) [inline app/models/post.rb]: can you add comment to say what this is for reference?

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 require 'rails_helper'
 
+# disabling tests as this is legacy code which we are not using
```
Comment (dct-lau17) [inline spec/models/post_spec.rb:3]: is there any unit tests to test when a user is logged in?

**Review by dct-lau17 — CHANGES_REQUESTED** [review]
only partly reviewed, will carry on in the morning

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -16,6 +16,8 @@ def create
 
   def show
     @user = User.find(session[:current_user_id])
+    @post = Post.new
+    @posts = Post.all
```
Comment (jeff1108) [inline app/controllers/users_controller.rb:20]: @posts = Post.new.all?????

File: app/models/user.rb [hunk 5]
```diff
@@ -3,8 +3,10 @@ class User < ApplicationRecord
   # This is a module, part of ActiveModel library
   # full details here: https://api.rubyonrails.org/classes/ActiveModel/SecurePassword/ClassMethods.html
   has_secure_password
+  has_many :posts
```
Comment (jeff1108) [inline app/models/user.rb]: How it works?

File: db/migrate/20180801142405_add_foreign_key_to_posts.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class AddForeignKeyToPosts < ActiveRecord::Migration[5.1]
+  def change
+    add_column :posts, :user_id, :integer
+    add_foreign_key :posts, :users
```
Comment (jeff1108) [inline db/migrate/20180801142405_add_foreign_key_to_posts.rb:4]: is it refer to :user_id?

File: db/schema.rb [hunk 7]
```diff
@@ -30,4 +31,5 @@
     t.datetime "updated_at", null: false
   end
 
+  add_foreign_key "posts", "users"
```
Comment (jeff1108) [inline db/schema.rb:34]: delete posts?

File: Gemfile (hunk 1 above)
Comment (elishkaflint) [reply Gemfile]: done

File: app/models/post.rb (hunk 2 above)
Comment (elishkaflint) [reply app/models/post.rb]: done

**Review by jeff1108 — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (elishkaflint) [reply app/controllers/users_controller.rb:20]: The view users/show.html.erb needs these 3 variables set up in the controller so they can be used. @post is needed for the post#create route in line 8 of that view.

File: app/models/user.rb (hunk 5 above)
Comment (elishkaflint) [reply app/models/user.rb]: see comment

File: db/migrate/20180801142405_add_foreign_key_to_posts.rb (hunk 6 above)
Comment (elishkaflint) [reply db/migrate/20180801142405_add_foreign_key_to_posts.rb:4]: as discussed

File: db/schema.rb (hunk 7 above)
Comment (elishkaflint) [reply db/schema.rb:34]: need both to establish the connection I think!

File: spec/models/post_spec.rb (hunk 3 above)
Comment (elishkaflint) [reply spec/models/post_spec.rb:3]: Took out that line as it was wrong! Don't think we need unit tests for Post as we haven't added any methods beyond in-built active record etc. Let's get the checker working and we may have to reassess!

**Comment by jeff1108** [comment]
:shipit:

**Review by jeff1108 — APPROVED** [review]

**Review by dct-lau17 — APPROVED** [review]

**Comment by dct-lau17** [comment]
:shipit:

**Pull request merged**